packages/hint-create-element-svg/src/hint.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ * @fileoverview Inform users that they need to use createElementNS to create SVG elements instead of createElement
+ */
+
+import { HintContext } from 'hint/dist/src/lib/hint-context';
+import { IHint, ElementFound, IAsyncHTMLElement, ProblemLocation } from 'hint/dist/src/lib/types';
+import { debug as d } from 'hint/dist/src/lib/utils/debug';
+import { ScriptEvents, ScriptParse } from '@hint/parser-javascript';
+import { findProblemLocation } from 'hint/dist/src/lib/utils/location-helpers';
+
+import { Linter } from 'eslint';
+import * as ESTree from 'estree';
+
+import meta from './meta';
+import svgElements from './svgElements';
+
+const debug: debug.IDebugger = d(__filename);
+
+export default class CreateElementSvgHint implements IHint {
+
+    public static readonly meta = meta;
+
+    /** Finds the approximative location in the page's HTML for a match in an element. */
+    public findProblemLocation(element: IAsyncHTMLElement, content?: string): Promise<ProblemLocation> {
+        return findProblemLocation(element, { column: 0, line: 0 }, content);
+    }
+
+    public constructor(context: HintContext<ScriptEvents>) {
+        const linter = new Linter();
+
+        /*
+         * Check to see if any call expression in the AST has a callee with
+         * a 'createElement' property. If yes, check if that call expression's
+         * first argument is an SVG element. In that case, report an error as
+         * the correct identifier property for creating SVG elements is
+         * 'createElementNS'
+         */
+        linter.defineRule('svg-create', {
+            create(eslintContext) {
+                return {
+                    CallExpression(node) {
+                        const exp = node as ESTree.CallExpression;
+                        const args = exp.arguments;
+                        const callee = exp.callee as ESTree.MemberExpression;
+                        const property = callee !== undefined && callee !== null ? callee.property as ESTree.Identifier : null;
+
+                        if (property !== null && property.name === 'createElement' && args.length >= 1 && args[0].type === 'Literal') {
+                            const arg = args[0] as ESTree.Literal;
+
+                            if (arg.value !== undefined && arg.value !== null && svgElements.has(arg.value.toString().toLowerCase())) {
+                                eslintContext.report({
+                                    messageId: 'avoidElement',
+                                    node: property
+                                });
+                            }
+                        }
+                    }
+                };
+            },
+            meta: { messages: { avoidElement: 'Avoid using createElement to create SVG elements; use createElementNS instead' } }
```

Comment:
Maybe make this a bit stronger since not only should you avoid it - it won't work.
```suggestion
            meta: { messages: { avoidElement: 'SVG elements cannot be created with createElement; use createElementNS instead' } }
```